client/src/app/pages/assessment/components/assessment-actions/components/dynamic-assessment-actions-row.tsx

```diff
@@ -129,18 +137,59 @@ const DynamicAssessmentActionsRow: FunctionComponent<
       }
     } catch (error) {
       console.error("Error while creating assessment:", error);
+      pushNotification({
+        title: t("terms.error"),
+        variant: "danger",
+      });
     }
   };
+
+  const takeAssessment = async () => {
+    if (!isArchetype && application?.archetypes?.length) {
+      for (const archetypeRef of application.archetypes) {
+        try {
+          const assessments = await getAssessmentsByItemId(
+            true,
+            archetypeRef.id
+          );
+
+          if (assessments && assessments.length > 0) {
+            setArchetypeRefToOverride(archetypeRef);
+            return;
```

Comment:
is this `return` needed or is it supposed to be a `break`?

Reply:
Whoops. Thanks good catch.